pkg/cmd/cli/cmd/edit.go

```diff
@@ -306,6 +339,37 @@ func (r *editResults) AddError(err error, info *resource.Info) string {
 	}
 }
 
+type patchError struct {
+	error
+}
+
+// tryPatch reads the latest version of the object, writes it to version, then attempts to merge
+// the changes onto it without conflict. If a conflict occurs jsonmerge.IsConflicting(err) is
+// true. The info object is mutated
+func tryPatch(delta *jsonmerge.Delta, info *resource.Info, version string) error {
```

Comment:
Hrmmm.  Interesting.  Although we endorse mutation already on Info, so I could be otherwise.

> On May 7, 2015, at 9:55 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cmd/edit.go:
> 
> > @@ -306,6 +339,37 @@ func (r *editResults) AddError(err error, info *resource.Info) string {
> >  	}
> >  }
> >  
> > +type patchError struct {
> > +	error
> > +}
> > +
> > +// tryPatch reads the latest version of the object, writes it to version, then attempts to merge
> > +// the changes onto it without conflict. If a conflict occurs jsonmerge.IsConflicting(err) is
> > +// true. The info object is mutated
> > +func tryPatch(delta *jsonmerge.Delta, info *resource.Info, version string) error {
> I'm not a fan of direct mutation. How would feel about returning a new *resource.Info?
> 
> —
> Reply to this email directly or view it on GitHub.
>